### PR TITLE
Implement iterator design pattern

### PR DESCRIPTION
**Pull request by 1239uth**
Might have to merge this AFTER merging 'Refactor authentication and gateways' since it contains those changes

### CODE REVIEW

File: app/src/main/java/com/example/fitappa/Authentication/DatabaseConstants.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ */
+public class DatabaseConstants {
+
+    /**
+     * Get the collection name for users
+     *
+     * @return String representing collection name for users
+     */
+    public String getUsersCollection() {
+        return "users";
```
Comment (sourenrex) [inline app/src/main/java/com/example/fitappa/Authentication/DatabaseConstants.java:14]: arent constants suppose to be all capital or is this just for data base

**Review by sourenrex — CHANGES_REQUESTED** [review]

**Review by sourenrex — APPROVED** [review]
looks good

**Review by sourenrex — APPROVED** [review]
looks really good

**Review by nxabdullah — APPROVED** [review]

**Comment by nxabdullah** [comment]
This is pretty good, well done

**Pull request merged**